topi/python/topi/vision/nms.py

```diff
@@ -278,7 +278,7 @@ def hybrid_nms(data, sorted_index, valid_count,
         # Only return max_output_size valid boxes
         num_valid_boxes = 0
         if max_output_size > 0:
-            for j in parallel(valid_count[i]):
+            for j in range(valid_count[i]):
```

Comment:
Does the issue come from the else branch ```num_valid_boxes += 1```?

Reply:
The logic just isn't truly parallel. As it stands, it returns a random selection of boxes which isn't the behaviour of the reference tflite operator (and I can't imagine is ever the intended behaviour). There may also be a race condition around num_valid_boxes, but I don't know enough about the implementation of hybrid script to confirm that.

Reply:
The parallel loop in hybrid script is the same as adding a parallel intrinsic to a loop axis in tvm schedule. Here num_valid_boxes will cause race condition.